src/core/scenario.jl

```diff
@@ -91,33 +102,39 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
                 if !(haskey(pv, "name"))
                     pv["name"] = string("PV", i)
                 end
-                push!(pvs, PV(;dictkeys_tosymbols(pv)...))
+                push!(pvs, PV(;dictkeys_tosymbols(pv)..., off_grid_flag = settings.off_grid_flag))
             end
         elseif typeof(d["PV"]) <: AbstractDict
             check_pv_tilt!(d["PV"], site)
-            push!(pvs, PV(;dictkeys_tosymbols(d["PV"])...))
+            push!(pvs, PV(;dictkeys_tosymbols(d["PV"])..., off_grid_flag = settings.off_grid_flag))
         else
             error("PV input must be Dict or Dict[].")
         end
     end
 
     if haskey(d, "Financial")
-        financial = Financial(; dictkeys_tosymbols(d["Financial"])...)
+        financial = Financial(; dictkeys_tosymbols(d["Financial"])..., off_grid_flag = settings.off_grid_flag )
     else
-        financial = Financial()
+        financial = Financial(; off_grid_flag = settings.off_grid_flag)
     end
 
-    if haskey(d, "ElectricUtility")
+    if haskey(d, "ElectricUtility") && !(settings.off_grid_flag)
         electric_utility = ElectricUtility(; dictkeys_tosymbols(d["ElectricUtility"])...)
-    else
+    elseif !(settings.off_grid_flag)
         electric_utility = ElectricUtility()
+    elseif settings.off_grid_flag 
+        if haskey(d, "ElectricUtility")
+            @warn "ElectricUtility inputs are not applicable when off_grid_flag is true and any ElectricUtility inputs will be ignored. For off-grid scenarios, a year-long outage will always be modeled."
+        end
+        electric_utility = ElectricUtility(; outage_start_time_step = 1, outage_end_time_step = settings.time_steps_per_hour * 8760) 
     end
 
     storage_structs = Dict{String, AbstractStorage}()
     if haskey(d,  "ElectricStorage")
         storage_dict = dictkeys_tosymbols(d["ElectricStorage"])
+        storage_dict[:off_grid_flag] = settings.off_grid_flag
     else
-        storage_dict = Dict(:max_kw => 0.0)
+        storage_dict = Dict(:max_kw => 0.0) # TODO: do we want to change to the off-grid inputs even if not modeling storage? 
```

Comment:
I don't understand the question for this TODO: what does it mean to "change to the off-grid inputs"?

Reply:
I see. No there is no need to update defaults for any item that is not modeled in the Julia package. The Julia package does not affect what is saved for inputs in the API database.